tool_quality_checks/db.py

```diff
@@ -0,0 +1,179 @@
+            if fetch is True:
+                return cur.fetchall()
+
+        except psycopg2.DatabaseError as e:
+            self._raise(e)
+
+    def select_table_names(self, search_table_name, schema=None):
+        schema = schema or self.schema
+
+        statement = """
+        SELECT table_name
+        FROM   information_schema.tables
+        WHERE  table_schema = '{schema}'
+        AND    table_name LIKE '{search_table_name}'
+        AND    table_type = 'BASE TABLE';
+        """.format(
+            schema=schema, search_table_name=search_table_name
+        )
+        return [i[0] for i in self.free_form(statement, fetch=True)]
+
+    def create_schema(self, schema_name, drop_schema=False):
+        """create a schema"""
+        if drop_schema == True:
+            drop_schema = """DROP SCHEMA IF EXISTS {schema_name} CASCADE;""".format(
+                schema_name=schema_name
+            )
+            self.free_form(drop_schema, fetch=False)
+        create_schema_statement = """
+        CREATE SCHEMA IF NOT EXISTS {schema_name}
+        ;""".format(
+            schema_name=schema_name
+        )
+        self.free_form(sql_statement=create_schema_statement, fetch=False)
+
+    def populate_geometry_columns(self):
+        """Populate geometry columns"""
+        populate_geometry_columns_statement = """
+        SELECT Populate_Geometry_Columns();"""
+        self.free_form(sql_statement=populate_geometry_columns_statement, fetch=False)
+
+    def perform_checks_with_sql(self, settings, check_table, check_type):
+        """
+        Performs quality checks on postgres DB
+
+        :param check_table - list of one or more structure tables (e.g. v2_manhole)
+        :param check_type - select type of check: completeness, quality
+        """
+        check_table = check_table.replace("v2_", "")
+        sql_template_name = "sql_" + check_type + "_" + check_table
+        if sql_template_name in sql_checks.sql_checks:
+            statement = sql_checks.sql_checks[sql_template_name].format(
+                **settings.__dict__
+            )
+            self.free_form(sql_statement=statement, fetch=False)
+
+    def create_view(self, view_table, view_schema, drop_view=True):
+        """
+        Creates a view with a join to v2_connection_nodes table
+        
+        :param view_table - table of which the view is created
+        """
+        if drop_view == True:
+            drop_statement = """DROP VIEW IF EXISTS {view_table};""".format(
+                view_table=view_table
+            )
+            self.free_form(drop_statement, fetch=False)
+        create_statement = sql_views.sql_views[view_table].format(schema=view_schema)
+        self.free_form(sql_statement=create_statement, fetch=False)
+
+    def execute_sql_file(self, filename):
+        # Open and read the file as a single buffer
+        fd = open(filename, "r")
+        sql_file = fd.read()
```

Comment:
Dit zou ik dan veranderen in:

```suggestion
        sql = sql_file.read()
```

want het is niet de sql file, maar de inhoud ervan.